compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureGenerator.java

```diff
@@ -843,6 +904,14 @@ public void visit(BLangListConstructorExpr.BLangArrayLiteral arrayLiteral) {
 
     @Override
     public void visit(BLangRecordLiteral recordLiteral) {
+        for (RecordLiteralNode.RecordField field : recordLiteral.fields) {
+            if (field.isKeyValueField()) {
```

Comment:
Don't we need to visit for other kinds of fields? They can be captured variables too?

Reply:
Updated. Please check.